### PR TITLE
Azure build for ubuntu 20.04 (et al.) utilizing clang

### PR DESCRIPTION
**Pull request by r0ost3r**
DONE: fix azure
~~TODO: implement std::swap~~ (*not a TODO here*)

https://stackoverflow.com/questions/5695548/public-friend-swap-member-function

### CODE REVIEW

**Comment by r0ost3r** [comment]
https://stackoverflow.com/questions/6380862/how-to-provide-a-swap-function-for-my-class

**Comment by Necktschnagge** [comment]
@r0ost3r I changed the base branch.
Please fix conflicts.

You cannot merger this into main as long as it builds upon my v1.1 engine which is not yet merged.

**Comment by Necktschnagge** [comment]
OK, I merged it by myself, all done.

File: CMakeLists.txt [hunk 1]
```diff
@@ -52,6 +52,11 @@ set(CMAKE_CXX_EXTENSIONS False)
 
 set(THREADS_PREFER_PTHREAD_FLAG ON)
 
+if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libstdc++ -std=c++20")
```
Comment (Necktschnagge) [inline CMakeLists.txt:56]: Is `-std=c++20` needed. We already set standard to 20.

Comment (r0ost3r) [reply CMakeLists.txt:56]: Yes, nevermind.

**Comment by Necktschnagge** [comment]
https://onlinegdb.com/vAPBcJGbJ

This seems to be a working minimal example for std::swap

**Comment by Necktschnagge** [comment]
I fixed the std::swap problem with
https://github.com/Necktschnagge/tobor-games/commit/dc5f57ed81fb766a9c977e95fb613edc6be1d6e8

You can ignore std::swap in this Pull request, just focus on the Azure / ubuntu 20 / clang stuff.

**Comment by Necktschnagge** [comment]
If you want to replace a file by its version of another commit (c16a5e3), you may also use
`git checkout {branch} path/to/file.cpp`
Just saying if you did not already know @r0ost3r

**Comment by r0ost3r** [comment]
@Necktschnagge 
azure fixed

**Review by Necktschnagge — APPROVED** [review]
OK.

**Pull request merged**